runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -2564,14 +3246,28 @@ TR::Node *TR_VectorAPIExpansion::transformNary(TR_VectorAPIExpansion *opt, TR::T
 
       for (int32_t i = 0; i < numOperands; i++)
          {
-         if (operands[i]->getOpCodeValue() == TR::aload)
+         TR::Node* operand = operands[i];
+
+         if (operand->getOpCodeValue() == TR::aload ||
+             operand->getOpCodeValue() == TR::acall)
```

Comment:
I think the code is simpler if you change this `if` statement to:

```
if (operand->getOpCodeValue() == TR::aload)
   {
   if (opt->_aliasTable[operandId]._objectType == Mask)
      {
      opCodeType = ...;
      }
   vectorizeLoadOrStore(opt, operand, opCodeType);
   }
else if (operand->getOpCodeValue() == TR::acall)
   {
   if (opt->_nodeTable[operand->getGlobalIndex()]._objectType == Mask)
      {
      opCodeType = ...;
      }
   vectorizeLoadOrStore(opt, operand, opCodeType);
   }
```

Reply:
Done.

Reply:
Added another comment.